server/controllers/public.js

```diff
@@ -61,11 +62,23 @@ exports.HomePageProjects = async (req, res) => {
 
 exports.GetProjectByID = async (req, res) => {
   const { id } = req.body;
-  
+
   try {
     const project = await Project.findById(id);
     return res.status(200).json(project);
   } catch (err) {
     return res.status(500).json({ error: "500 Internal Error" });
   }
 };
+
+exports.GetUserByID = async (req, res) => {
+  const { id } = req.body;
+  
+  try {
+    const user = await UserModel.findById(id);
+    return res.status(200).json(user);
```

Comment:
Add a conditional statement for user exist or not like - 
```diff
+ if(user)
+     return res.status(200).json(user)
+ else 
+     return res.status(404).json({ "error" : "user not found" })

Reply:
I have added this , Review it now